### DISCUSSION TITLE
createDataColumn gives error Argument of type 'string' is not assignable to parameter of type 'AccessorFn<unknown>'.ts(2345)

### DISCUSSION THREAD

**Discussion by fahadahmedln** [TanStack/table#4065] [Q&A]
I'm simply passing in a string as the first arg just like the docs show

```
table.createDataColumn("test", //error here
            { 
              id: field.FieldName,
              header: () => renderCellComponent(field),
              cell: (info: { getValue: () => string }) => info.getValue(),
            });
```

I get error: 

> Argument of type 'string' is not assignable to parameter of type 'AccessorFn<unknown>'.ts(2345)

**Comment by bkilrain** [comment]
I'm having the same issue with `@tanstack/react-table@8.0.0-beta.9` and `typescript@^4.7.4`.

**Reply by bkilrain** [reply]
Similar issue: https://github.com/TanStack/table/discussions/4054

**Comment by ShawnFarsai** [comment]
idk why this works, but the fix according to chatgpt, worked for me

<img width="895" alt="CleanShot 2023-04-14 at 11 47 08@2x" src="https://user-images.githubusercontent.com/24238821/232130999-c36499f6-edfa-4bd7-86d1-10d0c2ed68e2.png">

**Reply by brady-man-847** [reply]
it returns  (obj: unknown) => unknown

**Comment by dmitrisanzharov** [comment]
You need to add TYPE to your createColumnHelper, otherwise you will get this error:

GOOD: const columnHelper = createColumnHelper<any>();
BAD: const columnHelper = createColumnHelper(); // no type

**Reply by brentonmallen1** [reply]
What's the difference here? they look the same to me

**Comment by serhii-kucherenko** [comment]
> What's the difference here? they look the same to me

do this:

```
const columnHelper = createColumnHelper<MyRowType>();
```
